src/main/java/com/clevertap/stormdb/StormDB.java

```diff
@@ -571,6 +594,63 @@ public void flush() throws IOException {
         }
     }
 
+    private void readFromDeletedKeysFile(RandomAccessFile file, Consumer<Integer> entryConsumer) throws IOException{
+        ByteBuffer bufferDeletedKeys = ByteBuffer.allocate(4); // will hold 10 keys
+        while(true) {
+            bufferDeletedKeys.clear();
+            final int bytesRead = file.read(bufferDeletedKeys.array());
+            if (bytesRead == -1) {
+                break;
+            }
+            entryConsumer.accept(bufferDeletedKeys.getInt());
+        }
+    }
+
+    private TIntHashSet getKeysToBeDeletedFromFile(final boolean useLatestWalFile, final boolean readInMemoryBuffer) throws IOException{
+        TIntHashSet keysToBeDeleted = new TIntHashSet();
+
+        Consumer<Integer> entryConsumer = keyToBeDeleted -> {
+            int address = index.get(keyToBeDeleted);
+            if (address == RESERVED_KEY_MARKER) {
+                // delete only when key is not set in index
+                // because a set value will itself overwrite previous value so no need to delete
+                keysToBeDeleted.add(keyToBeDeleted);
+            }
+        };
+
+        final ArrayList<RandomAccessFile> filesToRead = new ArrayList<>(2);
```

Comment:
Is it possible to switch to a simple FileInputStream here? Random access files are a headache to deal with usually (in terms of buffering, etc).

Reply:
let me try this